addons/sourcemod/scripting/ttt/ttt_shop.sp

```diff
@@ -1456,3 +1581,138 @@ int GiveClientItem(int client, char[] sItem)
 
     return false;
 }
+
+bool IsItemInInventory(int client, const char[] itemshort)
+{
+	StringMap InventoryMap = playerInventory[client][hInvItems];
+	if (InventoryMap == INVALID_HANDLE)
+	{
+		LogToFile(g_sLog, "Function: IsItemInInventory - Inventory not intialised for client: %d.", client);
+		return false;
+	}
+	
+	int amount = 0;
+	bool exists = InventoryMap.GetValue(itemshort, amount);
+	
+	if (exists && amount > 0)
```

Comment:
Same thing here the item should simply be removed from the stringmap once i reaches an amount of 0. Therefore this can be written as "return InventoryMap.GetValue(itemshort, amount);". Saves some lines ^^